types/lodash/lodash-tests.ts

```diff
@@ -4062,29 +4062,52 @@ namespace TestAt {
     let array: AbcObject[] | null | undefined = [] as any;
     let list: _.List<AbcObject> | null | undefined = [] as any;
     let dictionary: _.Dictionary<AbcObject> | null | undefined = anything;
+    let numericDictionary: _.NumericDictionary<AbcObject> | null | undefined = anything;
+    let abcObject: AbcObject | null | undefined = anything;
 
     {
         let result: AbcObject[];
 
-        result = _.at<AbcObject>(array, 0, '1', [2], ['3'], [4, '5']);
-        result = _.at<AbcObject>(list, 0, '1', [2], ['3'], [4, '5']);
-        result = _.at<AbcObject>(dictionary, 0, '1', [2], ['3'], [4, '5']);
+        result = _.at(array, 0, '1', [2], ['3'], [4, '5']);
+        result = _.at(list, 0, '1', [2], ['3'], [4, '5']);
+        result = _.at(dictionary, 0, '1', [2], ['3'], [4, '5']);
+        result = _.at(numericDictionary, 0, '1', [2], ['3'], [4, '5']);
+    }
+
+    {
+        let result: Array<AbcObject[keyof AbcObject]>;
+
+        result = _.at(abcObject, 'a', ['b'], ['a', 'b']);
```

Comment:
Testing using `result = x` is an old way of doing it, and it's not very good because it only proves that `x` is *assignable to* `Array<AbcObject[keyof AbcObject]>`. It doesn't prove that the return type is *exactly* `Array<AbcObject[keyof AbcObject]>`. For example, `_.at()` could return `any`, and this test would still pass.

A better way is to use `$ExpectType`:

```ts
_.at(abcObject, 'a', ['b'], ['a', 'b']); // $ExpectType AbcObject[keyof AbcObject][]
```

The downside is that `dtslint` does a dumb string comparison when it sees `$ExpectType`, so your type needs to *exactly match* what `dtslint` is expecting, including whitespace. Also, you'll need to use `[]` instead of `Array<>`.

Also, since `$ExpectType` allows you to remove `let result: x;`, the braces (`{}`) are no longer necessary. i.e. instead of this:

```ts
{
    _.at(abcObject, 'a', ['b'], ['a', 'b']); // $ExpectType AbcObject[keyof AbcObject][]
}
```

just do this:

```ts
_.at(abcObject, 'a', ['b'], ['a', 'b']); // $ExpectType AbcObject[keyof AbcObject][]
```

Reply:
This comment also applies to tests below.

Reply:
Yeah; there were a couple bugs I found already that were masked by false positive test cases using  "assignability". I started updating some unit tests and writing new unit tests using $ExpectType comments later on. It's hard to decide when to follow the existing pattern vs when to rework to the better approach. Gotta be careful to avoid going down the rabbit hole that results in every unit test being rewritten and creating an impossibly gigantic PR to review.

I'll work on updating some more of these test cases. Let me know if you have any good advice on how far down the rabbit hole I should go.

Reply:
Agreed. In general, my approach is to clean up tests as I'm touching them. And if I create new ones, obviously create them using the better style. Probably don't worry about the tests where you only removed `<AbcObject>`, though.

Right now, actually, don't go to crazy updating the tests. I just found out that in order to support `ReadonlyArray<T>`, I'm going to need to do some major optimizations to the tests, which means I'll probably end up converting them all to the new style anyway.